extensions/scarb-snforge-test-collector/src/main.rs

```diff
@@ -0,0 +1,88 @@
+use anyhow::{Context, Result};
+use clap::Parser;
+use fs4::FileExt;
+use std::fs::{create_dir_all, File};
+use std::io::Write;
+
+use scarb_metadata::MetadataCommand;
+use scarb_ui::args::PackagesFilter;
+
+use crate::compilation::compile_tests;
+use crate::crate_collection::collect_test_compilation_targets;
+use crate::metadata::{
+    compilation_unit_for_package, corelib_for_compilation_unit, crates_config_for_compilation_unit,
+    dependencies_for_compilation_unit,
+};
+
+mod compilation;
+mod crate_collection;
+mod metadata;
+
+/// Starknet Foundry private extension for compiling test artifacts.
+/// Users should not call it directly.
+#[derive(Parser)]
+#[command(version)]
+struct Args {
+    #[command(flatten)]
+    packages_filter: PackagesFilter,
+}
+
+fn main() -> Result<()> {
+    let args = Args::parse();
+
+    let metadata = MetadataCommand::new().inherit_stderr().exec()?;
+    let selected_packages_metadata = args.packages_filter.match_many(&metadata)?;
+
+    let target_dir = metadata
+        .target_dir
+        .clone()
+        .unwrap_or_else(|| metadata.workspace.root.join("target"));
+    let snforge_target_dir = target_dir.join(&metadata.current_profile).join("snforge");
+    create_dir_all(&snforge_target_dir)?;
```

Comment:
```suggestion
   create_output_dir::create_output_dir(   path_to_target   )?;
    create_dir_all(&snforge_target_dir)?;
```
Please ensure that target dir is created with https://github.com/software-mansion/scarb/tree/main/utils/create-output-dir first.